analyzers/src/SonarAnalyzer.CSharp/Rules/PrivateFieldUsedAsLocalVariable.cs

```diff
@@ -229,19 +212,32 @@ public override void VisitIdentifierName(IdentifierNameSyntax node)
                     var pseudoStatement = GetParentPseudoStatement(memberReference);
                     if (pseudoStatement != null)
                     {
-                        var invocationsInPseudoStatement = this.invocations.GetOrAdd(pseudoStatement, x => new HashSet<ISymbol>());
+                        var invocationsInPseudoStatement = invocations.GetOrAdd(pseudoStatement, x => new HashSet<ISymbol>());
                         invocationsInPseudoStatement.Add(memberReference.Symbol);
                     }
                 }
             }
 
-            // A PseudoStatement is a Statement or an ArrowExpressionClauseSyntax (which denotes an expression-bodied member)
-            private static SyntaxNode GetParentPseudoStatement(SyntaxNodeWithSymbol<SyntaxNode, ISymbol> memberReference)
+            /// <summary>
+            /// Returns all statements before the specified statement within the containing method.
+            /// This method recursively traverses all parent blocks of the provided statement.
+            /// </summary>
+            private static IEnumerable<StatementSyntax> GetPreviousStatements(StatementSyntax statement)
```

Comment:
If that is possible, wait till https://github.com/SonarSource/sonar-dotnet/pull/4335 will be merged, since there I've moved this method to shared code.